src/types/relocatable.rs

```diff
@@ -129,12 +130,16 @@ impl MaybeRelocatable {
     ///Substracts two MaybeRelocatable values and returns the result as a MaybeRelocatable value.
     /// Only values of the same type may be substracted.
     /// Relocatable values can only be substracted if they belong to the same segment.
-    pub fn sub(&self, other: &MaybeRelocatable) -> Result<MaybeRelocatable, VirtualMachineError> {
+    pub fn sub(
+        &self,
+        other: &MaybeRelocatable,
+        prime: BigInt,
```

Comment:
Pass by reference.

Reply:
The rest of the functions in `relocatable.rs` are not being passed by reference. I opened an issue (#264)

Reply:
We can start by doing it here. Several has been fixed already in a PR (which you approved and I merged just now).

Reply:
Resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/247/commits/88a3e7720eac9cae9e5a8ac8bdbb1a4db0ee0bc8